### PR TITLE
Making debugger hit and bind

### PR DESCRIPTION
**Pull request by igor-ribeiiro**
Making breakpoints from the source code be hit again.
Fixes the problem where the debugger wouldn't be hit unless debugging directly.

This very probably wouldn't work on Mac, as it requires a direct reference to LocalAppData, which only exists in windows.

Pretty much we needed to directly reference the code that actually runs in the WEF cache.

### CODE REVIEW

**Comment by millerds** [comment]
Why can't we reference the mappings created by the project instead of something in the WEF cache.  We really should make this work everywhere and not just on Windows.

**Comment by igor-ribeiiro** [comment]
> Why can't we reference the mappings created by the project instead of something in the WEF cache. We really should make this work everywhere and not just on Windows.

The generated files/mapping in the `dist` folder don't get executed/touched. They simply are copied to the WEF cache and the ones in the WEF cache that actually get executed.
So we need to reference the files that actually get executed in some sort. I don't know where the WEF cache on mac is, but if the files that get executed are there, then we need some way to reference them

File: .vscode/launch.json [hunk 1]
```diff
@@ -51,10 +51,12 @@
       "type": "node",
       "request": "attach",
       "port": 9229,
-      "protocol": "inspector",
       "timeout": 600000,
       "preLaunchTask": "Debug: Excel Desktop",
-      "postDebugTask": "Stop Debug"
+      "postDebugTask": "Stop Debug",
+      "resolveSourceMapLocations": [
+        "${env:LOCALAPPDATA}/Microsoft/Office/16.0/Wef/**/*.js"
```
Comment (akrantz) [inline .vscode/launch.json:58]: So this tells VS Code debugger to pay attention to source maps that are in the WEF cache directory, right?

**Comment by millerds** [comment]
> > Why can't we reference the mappings created by the project instead of something in the WEF cache. We really should make this work everywhere and not just on Windows.
> 
> The generated files/mapping in the `dist` folder don't get executed/touched. They simply are copied to the WEF cache and the ones in the WEF cache that actually get executed. So we need to reference the files that actually get executed in some sort. I don't know where the WEF cache on mac is, but if the files that get executed are there, then we need some way to reference them

I don't think the source files have to be in the same directory as the executing javascript, so even if the js is coming from the wef the source file should be able to come from other places including the dist folder.  Did you try setting resolveSourceMapLocation to the dist folder?

File: .vscode/launch.json (hunk 1 above)
Comment (igor-ribeiiro) [reply .vscode/launch.json:58]: Yes, it will look to any source files there

**Comment by igor-ribeiiro** [comment]
> 

That's a nice idea that could work but didn't work in practice (I don't know if the source maps could be different).
But anyway, there is a default config for the `resolveSourceMapLocations` that makes it looks at all source maps, so it can also work on Mac. It does make the code less flaky

**Comment by igor-ribeiiro** [comment]
@akrantz can we get this reviewed again? @millerds is oof this week and would be nice to have this solved

**Comment by igor-ribeiiro** [comment]
@akrantz can we get this reviewed again? @millerds is oof this week and would be nice to have this solved

**Comment by akrantz** [comment]
The removal of "protocol: inspector" isn't really a needed part of this change. I'm not sure why it's being removed here.

Otherwise it looks good.

**Review by akrantz — CHANGES_REQUESTED** [review]
See other comments.

**Comment by igor-ribeiiro** [comment]
> The removal of "protocol: inspector" isn't really a needed part of this change. I'm not sure why it's being removed here.
> 
> Otherwise it looks good.

This protocol indeed has nothing to do with this bug, it was simply wrong there. The property `protocol` does not exist in `launch.json` when the request is `attach`.
This has written in the docs and also VSCode complains about this property as well, so I figured it would be a good practice to remove it

File: .vscode/launch.json [hunk 2]
```diff
@@ -51,10 +51,10 @@
       "type": "node",
       "request": "attach",
       "port": 9229,
-      "protocol": "inspector",
```
[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply .vscode/launch.json:54]: As my other comment, this property needs to be removed as is it is `not allowed`, if you want, I can create another PR with just the removal of the property as a change.

[the remark this answers is not in the document]
Comment (igor-ribeiiro) [reply .vscode/launch.json:54]: Created another PR to remove the property

**Review by akrantz — APPROVED** [review]

**Pull request merged**